### ISSUE TITLE
Setting memory_limit inside docker does not have an effect

### ISSUE THREAD

**Issue by michaeldiener** [tensorflow/tensorflow#60811]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Have you reproduced the bug with TF nightly?

No

### Source

source

### Tensorflow Version

2.12.0

### Custom Code

No

### OS Platform and Distribution

Ubuntu 22.04 (inside docker)

### Python version

3.10.6

### CUDA/cuDNN version

12.1

### GPU model and memory

Nvidia GeForce RTX 4090

### Current Behaviour?

Setting the memory_limit does not have any effect whatsoever when running inside a docker container.
I use the following NGC https://docs.nvidia.com/deeplearning/frameworks/tensorflow-release-notes/rel-23-05.html

### Standalone code to reproduce the issue

```shell
import tensorflow as tf

gpus = tf.config.experimental.list_physical_devices('GPU')
tf.config.set_logical_device_configuration(
	gpus[0],
	[tf.config.LogicalDeviceConfiguration(memory_limit=4096)]
)
```

### Relevant log output

_No response_</details>

**Comment by tilakrayal** [comment]
@michaeldiener,
The **memory_limit** in **tf.config.LogicalDeviceConfiguration** API mainly funtions for  Maximum memory (in MB) to allocate on the virtual device. Currently it is only supported for GPUs. 
Could you please confirm whether it is having the problem whenever you are running inside a docker container and also provide if there is any error log? Thank you!

**Comment by michaeldiener** [comment]
Thanks, didn't know about the model server. This might come handy eventually, but I need to be able to limit GPU memory while training in Jupyter inside docker. There is no way to do that? Again, setting set_logical_device_configuration does not have any effect.

**Comment by reedwm** [comment]
I cannot reproduce on a Titan V. I started the docker container with `docker run --privileged --gpus all -it --rm --shm-size=1g --ulimit memlock=-1 -p 8889:8889 nvcr.io/nvidia/tensorflow:23.05-tf2-py3`.

Then I ran the following program in the container to set the memory limit, initialize the GPU and print the memory usage with `nvidia-smi`

```python
import tensorflow as tf
import subprocess

gpus = tf.config.experimental.list_physical_devices('GPU')
tf.config.set_logical_device_configuration(
    gpus[0],
    [tf.config.LogicalDeviceConfiguration(memory_limit=4096)]
)

tf.constant(0.)  # Run GPU initialization
subprocess.run(['nvidia-smi'])
```

nvidia-smi showed only a bit more than 4096 MiB of memory being used.

@michaeldiener are there any extra steps needed to reproduce the issue for you? I doubt the fact I have a different GPU is causing `set_logical_device_configuration` to work correctly only for me.
